src/main/java/uk/gov/hmcts/darts/common/repository/TransformedMediaRepository.java

```diff
@@ -58,4 +59,30 @@ AND upper(tod.status.description) <> 'MARKED FOR DELETION'
         """)
     List<TransformedMediaEntity> findAllDeletableTransformedMedia(OffsetDateTime createdAtOrLastAccessedDateTime);
 
-}
+    @Query("""
+        SELECT tm
+            FROM TransformedMediaEntity tm
+            JOIN tm.mediaRequest media
+            JOIN media.hearing hearing
+            JOIN hearing.courtCase courtCase
+            JOIN hearing.courtroom courtroom
+            JOIN courtroom.courthouse courthouse
+        WHERE
+           (:mediaId IS NULL OR (:mediaId IS NOT NULL AND media.id=:mediaId)) AND
+           (:caseNumber IS NULL  OR (:caseNumber IS NOT NULL AND courtCase.caseNumber=:caseNumber)) AND
+           (:courtHouseDisplayName IS NULL OR (:courtHouseDisplayName IS NOT NULL AND courthouse.displayName like CONCAT('%', :courtHouseDisplayName, '%'))) AND
```

Comment:
is courtHouseDisplayName being populated from a dropdown or free text on the screens? i.e. does this need to be a case-insensitive search?

Reply:
Sorry @chrisbellingham-hmcts ? The requirements didnt seem to mention case insensitive but happy to add it

Reply:
Case-insensitive matching would be a good idea here (and also for "owner" and "requestedBy" fields), and would make behaviour align with user search. I've updated the LLD to include a case-insensitive requirement, if you wouldn't mind adding an AC to the ticket for it @mestebanez?